### ISSUE TITLE
bug: uninitialized MPI_Status object

### ISSUE THREAD

**Issue by mpichbot** [pmodels/mpich#2332]
**Originally by _jhammond_ on 2016-03-28 13:14:51 -0500**

---
# Message 1```

I notice an uninitialized MPI_Status object can make MPI_Get_count return a wrong result,
when the data amount in the MPI operation is zero (although I only checked MPI-IO).
Attached is a test program that uses MPI collective read where only root process has
non-zero length data to read. The expected result from MPI_Get_count is 0 for all
non-root process. To mimic an uninitialized MPI_Status object, I call memset to make
the object non-zero.

Here is the code fragment.

```
if (rank =# 0) len10;
else len = 0;

MPI_File_read_all(fh, buf, len, MPI_BYTE, &status);

MPI_Get_count(&status, MPI_BYTE, &get_size);
```

For process rank > 0, get_size may not be 0.

```
```
# Message 2```

My test program can be compiled with command "mpicc get_count.c -o get_count"
and run "mpiexec -n 4 get_count". For correct (expected) result, nothing
will be printed on stdout. Otherwise, error messages will be printed.

My point is MPI_Get_count does not report correct result because
MPI collective I/O call fails to initialize the MPI_Status object.

I found that OpenMPI and earlier version of MPICH (2-1.2.1) can run
this test code correctly.

The test failed when I ran MPICH 3.1.4 and the current from GIT repo.

```
```

**Comment by wkliao** [comment]
I am wondering the status of this issue.

**Comment by raffenet** [comment]
This was fixed in 4b7d553b4cbcd54ec6f4a6365871db267aac63ea.
